Bindings/Packages/com.virtualmaker.bindings/Runtime/Property.cs

```diff
@@ -259,4 +261,26 @@ public static DerivedProperty<TDerived> From<TValue, TDerived>(IProperty<TValue>
             return DerivedProperty<TDerived>.From<TValue>(property, func);
         }
     }
+
+    public static class IPropertyAwaitingExtensions
+    {
+        public static async Task WaitUntilAsync<T>(this IProperty<T> prop, Func<T, bool> predicate, CancellationToken cancellationToken = default)
```

Comment:
how is cancellationToken used?

Reply:
Was thinking it could be good to use it on the Task, like with Util.Async's to do `await tcs.Task.WithCancellation(cancellationToken).ConfigureAwait(true);`